packages/lesswrong/components/posts/AddToCalendar/AddToCalendarButton.tsx

```diff
@@ -0,0 +1,162 @@
+import { registerComponent, Components } from '../../../lib/vulcan-lib';
+import React, { useEffect, useState } from 'react';
+import moment from '../../../lib/moment-timezone';
+import { useTracking } from "../../../lib/analyticsEvents";
+import { useSingle } from '../../../lib/crud/withSingle';
+import makeUrls from './makeUrls';
+import classNames from 'classnames';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    position: 'relative',
+    textAlign: 'left'
+  },
+  button: {
+    display: 'inline-flex',
+    background: 'transparent',
+    color: theme.palette.grey[600],
+    font: 'inherit',
+    fontSize: 14,
+    verticalAlign: 'text-bottom',
+    '&:hover': {
+      opacity: 0.5
+    }
+  },
+  icon: {
+    height: 16,
+    fill: theme.palette.grey[600]
+  },
+  label: {
+    marginLeft: 8
+  },
+  dropdown: {
+    position: 'absolute',
+    left: 6,
+    background: '#FFF',
+    color: theme.palette.grey[700],
+    padding: '4px 0',
+    borderRadius: 4,
+    boxShadow: "0 1px 3px rgba(0, 0, 0, 0.2)",
+    zIndex: 1
+  },
+  option: {
+    display: 'block',
+    whiteSpace: 'nowrap',
+    padding: '6px 12px',
+  }
+})
+
+const AddToCalendarIcon = ({className=''}) => {
+  return <svg viewBox="207.59 11.407 18.981 20.638" xmlns="http://www.w3.org/2000/svg" className={className}>
+    <path d="M 225.59 21.407 L 225.59 15.407 C 225.59 14.307 224.69 13.407 223.59 13.407 L 222.59 13.407 L 222.59 11.407 L 220.59 11.407 L 220.59 13.407 L 212.59 13.407 L 212.59 11.407 L 210.59 11.407 L 210.59 13.407 L 209.59 13.407 C 208.49 13.407 207.59 14.307 207.59 15.407 L 207.59 29.407 C 207.59 30.507 208.49 31.407 209.59 31.407 L 216.59 31.407 L 216.59 29.407 L 209.59 29.407 L 209.59 19.407 L 223.59 19.407 L 223.59 21.407 L 225.59 21.407 Z"/>
+    <path d="M 226.571 27.045 L 226.571 29.045 L 223.571 29.045 L 223.571 32.045 L 221.571 32.045 L 221.571 29.045 L 218.571 29.045 L 218.571 27.045 L 221.571 27.045 L 221.571 24.045 L 223.571 24.045 L 223.571 27.045 L 226.571 27.045 Z"/>
+  </svg>
+}
+
+const AddToCalendarButton = ({post, label, hideTooltip, hideIcon, iconClassName, classes}: {
```

Comment:
I renamed this from `AddToCalendarIcon` - most of this component is the same as before